code/code/misc/utility.cc

```diff
@@ -1677,29 +1677,25 @@ bool TRoom::roomIsEmpty(bool ignore_imms) const
 
 void TBeing::giveMoney(TBeing *ch, int money, moneyTypeT type)
 {
-  if(money < 0){
+  if (money < 0) {
     vlogf(LOG_BUG, format("%s just tried to give negative money (%i) to %s") %
 	  getName() % money % ch->getName());
     return;
   }
     
-
   addToMoney(-money, type);
   ch->addToMoney(money, type);
 }
 
-
 void TBeing::addToMoney(int money, moneyTypeT type, bool allowTithe)
 {
   int lev = 0;
-  int amount;
   TDatabase db(DB_SNEEZY);
 
-  if(money && !bootTime)
-    vlogf(LOG_SILENT, format("%s talens changed by %i.") %
-	  getName() % money);
+  if (money && !bootTime)
+    vlogf(LOG_SILENT, format("%s talens changed by %i.") % getName() % money);
 
-  points.money += money;
+  setMoney(getMoney() + money);
```

Comment:
setMoney clamps to a zero minimum on its own, I felt this was better than redundantly adding additonal clamps.

Reply:
(deleted previous comment) addToMoney called with a negative amount is the standard method of removing money from someone's wallet, and happens all the time. I think pretty much all of the addToXXX methods are used with negative values along with positive ones.

Reply:
I decided on the setmoney/getmoney pair because setmoney warns and clamps to zero on negative values, and I thought adding a second warning/clamp here was redundant compared to just re-using the code path.